distributed/deploy/cluster.py

```diff
@@ -170,9 +173,27 @@ def sync(self, func, *args, asynchronous=None, callback_timeout=None, **kwargs):
         else:
             return sync(self.loop, func, *args, **kwargs)
 
+    def _log(self, log):
+        """Log a message.
+
+        Output a message to the user and also store for future retrieval.
+
+        For use in subclasses where initialisation may take a while and it would
+        be beneficial to feed back to the user.
+
+        Examples
+        --------
+        >>> self._log("Submitted job X to batch scheduler")
+        """
+        self._cm_logs.append((datetime.datetime.now(), log))
+        if not self.quiet:
+            print(log)
+
     async def _get_logs(self, scheduler=True, workers=True):
         logs = Logs()
 
+        logs["Cluster"] = Log("\n".join(line[1] for line in self._cm_logs))
```

Comment:
We don't want to include the timestamps in the output somehow?

Reply:
I'm not sure. I made a point to store them, but I don't think the scheduler or worker logs include them.